pkg/controller/clusterserviceversion/clusterserviceversion_controller.go

```diff
@@ -79,14 +91,35 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 				return !(watchOk && watchLabel == "lite")
 			},
 			DeleteFunc: func(evt event.DeleteEvent) bool {
-				return false
+				_, okAllNamespace := evt.Meta.GetLabels()[allnamespaceTag]
+				watchLabel, watchOk := evt.Meta.GetLabels()[watchTag]
+				_, ignoreOk := evt.Meta.GetAnnotations()[ignoreTag]
+
+				if strings.Contains(evt.Meta.GetName(), utils.CSV_NAME) {
```

Comment:
Need to update this to look for our meter definition annotation and queue if it's present.